icepyx/core/is2ref.py

```diff
@@ -330,13 +332,13 @@ def gt2spot(gt, sc_orient):
             elif gr_lr == "r":
                 spot = 6
 
-    if "spot" not in locals():
```

Comment:
Looking at `locals()` is a smell to me. I feel it may throw off some other code-readers.

Reply:
To confirm my understanding: new line 299 `spot: Optional[int] = None` will leave `spot` and it's value alone if it exists, otherwise create it and set it to `None`?

Reply:
`spot: Optional[int] = None` sets `spot` to `None` and annotates it as being optional (can take the value of `None`, which it is initialized to) or `int`. 

https://docs.python.org/3/library/typing.html#typing.Optional

> `Optional[X]` is equivalent to `X | None` (or `Union[X, None]`).

In our case, the equivalent would be `int | None`.

Reply:
Thanks for the link and explanation. My concern here then is that if there is already a variable named `spot`, we're overwriting it here so it will always be None. We'll need to move this line above the functionality that's trying to fill in the spot value.

Reply:
Yes, in this case we initialize `spot` to `None` (line 299), and then try to set it (lines 300-333). If, after that, `spot` is still `None` (line 335), we raise a `ValueError` (line 336). 

It boils down to this:

```
spot = None
<code to attempt setting spot to an integer>
if spot is None:  # `spot` is still `None` , the intervening code failed
    raise ValueError()
```

Reply:
Facepalm (and thanks for kindly re-explaining). Apparently when I last looked at this I missed that this conversation is happening a few lines later than the `spot: Optional[int] = None` line and instead thought the setting `spot=None` was happening after our `<code to attempt setting spot to an integer>`. Sorry about that!